source/common/http/conn_manager_impl.cc

```diff
@@ -1934,7 +1935,7 @@ void ConnectionManagerImpl::ActiveStream::addEncodedData(ActiveStreamEncoderFilt
   } else if (state_.filter_call_state_ & FilterCallState::EncodeTrailers) {
     // In this case we need to inline dispatch the data to further filters. If those filters
     // choose to buffer/stop iteration that's fine.
-    encodeData(&filter, data, false, FilterIterationStartState::AlwaysStartFromNext);
+    encodeData(&filter, data, end_stream, FilterIterationStartState::AlwaysStartFromNext);
```

Comment:
I think we need handling this correctly in encodeData:
https://github.com/envoyproxy/envoy/blob/defea7ecbf6f3ebffe8b9d41afa7f62322756f05/source/common/http/conn_manager_impl.cc#L1983

Also, some ASSERT needed when end_stream is added, decodeTrailers should only return StopIteration.

Reply:
>  (*entry)->end_stream_ = end_stream && !response_trailers_; 

@lizan, sorry, so previously it is always false, so we want to preserve that?

Reply:
Since !response_trailers is always `false` there in this case, the filters in front of grpc-web still get called encodeData(data, false) while they should get encodeData(data, true).